src/unix/cygwin/mod.rs

```diff
@@ -0,0 +1,3137 @@
+pub const TCP_KEEPIDLE: c_int = 3;
+pub const TCP_MAXSEG: c_int = 4;
+pub const TCP_QUICKACK: c_int = 12;
+pub const TCP_USER_TIMEOUT: c_int = 14;
+pub const TCP_FASTOPEN: c_int = 15;
+pub const TCP_KEEPCNT: c_int = 16;
+pub const TCP_KEEPINTVL: c_int = 17;
+
+pub const WINDOWS_POST: Ioctl = 0;
+pub const WINDOWS_SEND: Ioctl = 1;
+pub const WINDOWS_HWND: Ioctl = 2;
+
+pub const MOUNT_TEXT: c_uint = 0x01;
+pub const MOUNT_SYSTEM: c_uint = 0x08;
+pub const MOUNT_EXEC: c_uint = 0x10;
+pub const MOUNT_CYGDRIVE: c_uint = 0x20;
+pub const MOUNT_CYGWIN_EXEC: c_uint = 0x40;
+pub const MOUNT_SPARSE: c_uint = 0x80;
+pub const MOUNT_NOTEXEC: c_uint = 0x100;
+pub const MOUNT_DEVFS: c_uint = 0x200;
+pub const MOUNT_PROC: c_uint = 0x400;
+pub const MOUNT_RO: c_uint = 0x1000;
+pub const MOUNT_NOACL: c_uint = 0x2000;
+pub const MOUNT_NOPOSIX: c_uint = 0x4000;
+pub const MOUNT_OVERRIDE: c_uint = 0x8000;
+pub const MOUNT_IMMUTABLE: c_uint = 0x10000;
+pub const MOUNT_AUTOMATIC: c_uint = 0x20000;
+pub const MOUNT_DOS: c_uint = 0x40000;
+pub const MOUNT_IHASH: c_uint = 0x80000;
+pub const MOUNT_BIND: c_uint = 0x100000;
+pub const MOUNT_USER_TEMP: c_uint = 0x200000;
+pub const MOUNT_DONT_USE: c_uint = 0x80000000;
+
+pub const _POSIX_VDISABLE: cc_t = 0;
+
+pub const GRND_NONBLOCK: c_uint = 0x1;
+pub const GRND_RANDOM: c_uint = 0x2;
+
+pub const _IONBF: c_int = 2;
+pub const BUFSIZ: c_int = 1024;
+
+pub const POSIX_SPAWN_RESETIDS: c_int = 0x01;
+pub const POSIX_SPAWN_SETPGROUP: c_int = 0x02;
+pub const POSIX_SPAWN_SETSCHEDPARAM: c_int = 0x04;
+pub const POSIX_SPAWN_SETSCHEDULER: c_int = 0x08;
+pub const POSIX_SPAWN_SETSIGDEF: c_int = 0x10;
+pub const POSIX_SPAWN_SETSIGMASK: c_int = 0x20;
+
+// intentionally not public, only used for fd_set
+cfg_if! {
+    if #[cfg(target_pointer_width = "32")] {
+        const ULONG_SIZE: usize = 32;
+    } else if #[cfg(target_pointer_width = "64")] {
+        const ULONG_SIZE: usize = 64;
+    } else {
+        // Unknown target_pointer_width
+    }
+}
+
+f! {
+    pub fn FD_CLR(fd: c_int, set: *mut fd_set) {
+        let fd = fd as usize;
+        let size = core::mem::size_of_val(&(*set).fds_bits[0]) * 8;
+        (*set).fds_bits[fd / size] &= !(1 << (fd % size));
+    }
+
+    pub fn FD_ISSET(fd: c_int, set: *const fd_set) -> bool {
+        let fd = fd as usize;
+        let size = core::mem::size_of_val(&(*set).fds_bits[0]) * 8;
+        ((*set).fds_bits[fd / size] & (1 << (fd % size))) != 0
+    }
+
+    pub fn FD_SET(fd: c_int, set: *mut fd_set) {
+        let fd = fd as usize;
+        let size = core::mem::size_of_val(&(*set).fds_bits[0]) * 8;
+        (*set).fds_bits[fd / size] |= 1 << (fd % size);
+    }
+
+    pub fn FD_ZERO(set: *mut fd_set) {
+        for slot in (*set).fds_bits.iter_mut() {
+            *slot = 0;
+        }
+    }
+
+    pub fn CPU_ALLOC_SIZE(count: c_int) -> size_t {
+        let _dummy: cpu_set_t = core::mem::zeroed();
+        let size_in_bits = 8 * core::mem::size_of_val(&_dummy.bits[0]);
```

Comment:
`core::mem::zeroed()` is advised against nowadays. Could this just be `cpu_set_t { bits: [0; 16] }`?

Applies a couple of other places.

Reply:
The linux targets use `zeroed`.